CppCoreGuidelines.md

```diff
@@ -10077,14 +10077,18 @@ If at all possible, reduce the conditions to a simple set of alternatives (e.g.,
 
 ##### Example
 
-    bool owned = false;
-    owner<istream*> inp = [&]{
+    unique_ptr<istream> owner;
+    auto& in = [&] {
         switch (source) {
-        case default:       owned = false; return &cin;
-        case command_line:  owned = true;  return new istringstream{argv[2]};
-        case file:          owned = true;  return new ifstream{argv[2]};
+            case command_line:  owner = new istringstream{argv[2]}; return *owner;
```

Comment:
This is ill-formed (Also occurs on 10084):
* a raw pointer cannot be assigned to a `unique_ptr`
* `*owner` (an `istream`) cannot be copied into the return value of this function since `istream`s are not copyable